### ISSUE TITLE
A11Y: DataTable - Unable to tab through columns

### ISSUE THREAD

**Issue by rob-richards** [IBM/carbon-components-react#1780]
## Detailed description

> Current Behavior

You press tab into a row and start on column 2: the name, then click tab again and are in the overflow menu, passing right over all other columns. Arrow keys do not work at all.

> Desired Behavior

You should be able to navigate through the columns with the tab or arrow keys.

This issue can be seen on [storybook](http://react.carbondesignsystem.com/?selectedKind=DataTable&selectedStory=default&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel).

**Comment by asudoh** [comment]
Hi 👋 the content of data table can be controlled by consumers. e.g. You can put e.g. `<TableCell tabIndex="0">` in the `render` prop. Does it make sense...? Thanks!

**Comment by rob-richards** [comment]
Hey @asudoh. Yes, I have used `tabIndex={0}` on all `<TableCell>` components in our app. However, I thought this accessibility support should come from Carbon by default, which is why I created this issue.
